src/main/java/ca/fxco/api/pistonlib/pistonLogic/controller/PistonController.java

```diff
@@ -0,0 +1,166 @@
+package ca.fxco.api.pistonlib.pistonLogic.controller;
+
+import ca.fxco.api.pistonlib.pistonLogic.structure.StructureRunner;
+import ca.fxco.pistonlib.pistonLogic.families.PistonFamily;
+import ca.fxco.pistonlib.pistonLogic.structureResolvers.BasicStructureResolver;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.block.state.properties.PistonType;
+import net.minecraft.world.level.gameevent.GameEvent;
+
+/**
+ * An interface used as the base of all piston related calls from blocks.
+ * Having this logic split from the physical blocks allows any block to gain the ability to become a piston.
+ *
+ * @author FX
+ * @since 1.0.4
+ */
+public interface PistonController {
+
+    /**
+     * Gets the piston family used by this controller.
+     *
+     * @return The piston family for this controller.
+     */
+    PistonFamily getFamily();
+
+    /**
+     * Gets the piston type used by this controller.
+     *
+     * @return The piston type for this controller.
+     */
+    PistonType getType();
+
+    /**
+     * Sets the base block for the piston family.
+     * Should call {@link PistonFamily#setBase}
+     *
+     * @param block The block linked to this controller.
+     */
+    default void setBase(Block block) {
```

Comment:
I mentioned this briefly on Discord, but I think it'd be better if the families are bootstrapped after blocks, so the relevant blocks can be passed into the family's constructor directly, and then force blocks to access the family through the getter only (instead of through a final field like it is in some cases now).